src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -222,6 +233,19 @@ private SafeCurlMultiHandle CreateAndConfigureMultiHandle()
                         Interop.Http.CURLMoption.CURLMOPT_PIPELINING,
                         (long)Interop.Http.CurlPipe.CURLPIPE_MULTIPLEX));
                 }
+
+                // Configure max connections per host if it was changed from the default
+                if (_associatedHandler.MaxConnectionsPerServer < int.MaxValue) // int.MaxValue considered infinite, mapping to libcurl default of 0
```

Comment:
No.  It can't be changed after any connection has been attempted using the handler.  In general that's true for all properties of the handler: they can be changed until the handler has been used to initiate the first connection, after which they're all frozen and throw if you try to mutate them.